workitem/workitem_repository.go

```diff
@@ -690,48 +693,172 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
 	return &result, nil
 }
 
-// GetCountsPerIteration fetches WI count from DB and returns a map of iterationID->WICountsPerIteration
-// This function executes following query to fetch 'closed' and 'total' counts of the WI for each iteration in given spaceID
-// 	SELECT iterations.id as IterationId, count(*) as Total,
-// 		count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed
-// 		FROM "work_items" left join iterations
-// 		on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb
-// 		WHERE (iterations.space_id = '33406de1-25f1-4969-bcec-88f29d0a7de3'
-// 		and work_items.deleted_at IS NULL) GROUP BY IterationId
+// GetCountsPerIteration counts WIs including iteration-children and returns a map of iterationID->WICountsPerIteration
 func (r *GormWorkItemRepository) GetCountsPerIteration(ctx context.Context, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
+	db := r.db.Model(&iteration.Iteration{}).Where("space_id = ?", spaceID)
+	if db.Error != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	var allItrations []uuid.UUID
+	db.Pluck("id", &allItrations)
```

Comment:
thank you, done in https://github.com/almighty/almighty-core/pull/1151/commits/fb439257f1ee09bebdf4b186fb15cbda9824855a